Clarify the usage of regular expressions in included/excluded job lists in DD Jenkins plugin (#21378)
DataDog/documentation: content/en/continuous_integration/pipelines/jenkins.md

**Environment variable**: `DATADOG_JENKINS_PLUGIN_INCLUDED`<br/>
**Example**: `susans-job,johns-.*,prod_folder/prod_release`

> Keep in mind that the lists of included and excluded jobs can contain regular expressions and not glob patterns (so in order to include a job with a specific prefix you need to use `prefix-.*` and not `prefix-*`).

Reviewer: ```suggestion
Lists of included and excluded jobs can contain regular expressions, but not glob patterns. To include a job with a specific prefix, use `prefix-.*`—not `prefix-*`.
```
